include/swift/SIL/SILNodes.def

```diff
@@ -613,8 +625,14 @@ NON_VALUE_INST(AllocGlobalInst, alloc_global,
 NON_VALUE_INST(CondFailInst, cond_fail,
                SILInstruction, MayHaveSideEffects, DoesNotRelease)
 
-NODE_RANGE(SILInstruction, AllocStackInst, CondFailInst)
-NODE_RANGE(SILNode, SILPHIArgument, CondFailInst)
+NODE_RANGE(NonValueInstruction, UnreachableInst, CondFailInst)
+
+MULTIPLE_VALUE_INST(DestructureValueInst, destructure_value,
+                    SILInstruction, None, DoesNotRelease)
```

Comment:
Everything else in SIL draws a line between structs and tuples, even if that seems a little unnecessary; I'm not sure why this is an exception.

Reply:
This is not true when it comes to aggregates (i.e. the *_value instructions). You can perform a retain_value of a tuple or a struct for instance.

Reply:
Yes, that's true, but only because those operations don't care about the type structure at all.  A better comparison would be the dual operations here, "tuple" and "struct".